api/app/controllers/spree/api/v2/storefront/taxons_controller.rb

```diff
@@ -0,0 +1,44 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class TaxonsController < ::Spree::Api::V2::BaseController
+          def index
+            render json: serialize_taxons_collection(collection), status: 200
+          end
+
+          def show
+            render json: serialize_taxon_resource(resource), status: 200
+          end
+
+          private
+
+          def dependencies
+            {
+              taxons_finder:                Spree::Taxons::Find,
+              taxon_resource_serializer:    Spree::V2::Storefront::TaxonSerializer,
+              taxons_collection_serializer: Spree::V2::Storefront::TaxonSerializer
+            }
+          end
+
+          def collection
+            @taxons = Spree::Taxon.includes(:parent, :children).accessible_by(current_ability, :read)
+            @taxons = dependencies[:taxons_finder].new(@taxons, params).call
```

Comment:
Why do we need instance variables here? As far as I understand they're not used anywhere
Besides - a new instance of controller is created on each request therefore instance variables and memoization make no sense in this context.

Reply:
Agreed.